lib/facter/util/wmi.rb

```diff
@@ -2,7 +2,10 @@ module Facter::Util::WMI
   class << self
     def connect(uri = wmi_resource_uri)
       require 'win32ole'
-      WIN32OLE.connect(uri)
+      #WIN32OLE.connect(uri)
+      swbem = WIN32OLE.new("WbemScripting.SWbemLocator")
```

Comment:
In my testing, I'm not seeing a failure to call `OleInitialize` from Ruby:


```ruby
require 'ffi'

module COM
  extend FFI::Library

  ffi_convention :stdcall

  S_OK = 0
  S_FALSE = 1

  def SUCCEEDED(hr) hr >= 0 end
  def FAILED(hr) hr < 0 end

  module_function :SUCCEEDED, :FAILED

  def raise_if_hresult_failed(name, *args)
    failed = FAILED(result = send(name, *args)) and raise "#{name} failed (hresult #{format('%#08x', result)})."

    result
  ensure
    yield failed if block_given?
  end

  module_function :raise_if_hresult_failed

  # HRESULT CoInitialize(
  #   _In_opt_  LPVOID pvReserved
  # );
  ffi_lib :ole32
  attach_function :CoInitialize, [:pointer], :int32

  ffi_lib :ole32
  attach_function :OleInitialize, [:pointer], :int32

  # void CoUninitialize(void);
  ffi_lib :ole32
  attach_function :CoUninitialize, [], :void

  ffi_lib :ole32
  attach_function :OleUninitialize, [], :void


  def InitializeCom
    raise_if_hresult_failed(:CoInitialize, FFI::Pointer::NULL)

    puts 'Successfully called CoInitialize'

    at_exit { CoUninitialize() }
  end

  module_function :InitializeCom


  def InitializeOle
    raise_if_hresult_failed(:OleInitialize, FFI::Pointer::NULL)

    puts 'Successfully called OleInitialize'


    at_exit { OleUninitialize() }
  end

  module_function :InitializeOle
end

COM.InitializeOle
```

Results

```
[10.0.16.145]: PS C:\puppet> bundle exec ruby .\ole-init-test.rb
Successfully called OleInitialize
```